javascript/javascript2/week1/index.js

```diff
@@ -0,0 +1,141 @@
+const btnSubmit = document.querySelector("button[type=submit]");
+const btnReset = document.querySelector("button[type=reset]");
+    const spiritAnimals = [{
+        animal: "Alligator",
+        description: "Alligators are patient creatures, laying in wait to snap their jaws shut on their prey."
+        }, {
+        animal: "Ant",
+        description: "Ants are very goal-oriented and work well in groups."
+    }, {
+        animal: "Bat",
+        description: "Despite hanging upside down in caves, bats are quite resilient animals, using echolocation and their sensitive ears to communicate."
+    }, {
+        animal: "Bear",
+        description: "Bears aren’t the aggressive creatures many say they are; rather, they are nurturing and gentle. "
+    }, {
+        animal: "Bee",
+        description: "Bees are all about productivity and working hard to serve their queen."
+    }, {
+        animal:"Cat",
+        description:"If you keep seeing a cat or have a cat as your spirit animal, it means you need to take action, or it’s a negative warning of the future."
+    }, {
+        animal:"Deer",
+        description:"Deer are incredibly gentle, graceful animals whom we often see grazing meadows and protecting their offspring."
+    }, {
+        animal:"Dog",
+        description:"To have one as your spirit animal is a warning to be aware of hidden issues you haven’t come to know just yet; it’s also a reminder to possess a high level of loyalty with the people in your life, just like the animal does."
+    }, {
+        animal:"Elephant",
+        description:"Not only do elephants never forget, but they are highly intelligent mammals who symbolize honor, royalty, strength and power. "
+    }, {
+        animal:"Frog",
+        description:"We usually see frogs hanging out by bodies of water, and symbolize cleansing, rebirth, and peace."
+    }];
+
+
+let inputValue = '';
+
+input.addEventListener("input", onInputPress);
+btnSubmit.addEventListener("click", getRandomSpiritAnimal, { once: true });
+btnReset.addEventListener("click", getNewRandomSpiritAnimal);
+
+
+function onInputPress(event) {
+    // inputValue = event.target.value;
+    inputValue = input.value;
+    console.log(inputValue);
+  return inputValue;
+};
+
+
+function getRandomSpiritAnimal() {
+    if (!inputValue) {
+        return output.textContent = "Need the name";
+    };
+    const randNum = Math.floor(Math.random() * 10) + 1;
+    const randomAnimal = spiritAnimals[randNum];
+    const { animal, description } = randomAnimal;
+    output.textContent = ` ${inputValue} - ${animal}. ${description}`;
+    
+};
+
+function getNewRandomSpiritAnimal() {
+     if (!inputValue) {
+        return output.textContent = "Need the name";
+    };
+    const randNum = Math.floor(Math.random() * 10) + 1;
+    const randomAnimal = spiritAnimals[randNum];
+    const { animal, description } = randomAnimal;
+    output.textContent = ` ${inputValue} - ${animal}. ${description}`;
+};
```

Comment:
What is the difference between these two functions? Could it be that just one function is enough for both setting and resetting the spirit animal name? 

The more code you reuse, the less code you need to maintain (in a real life application ;) )